feature/client/src/main/java/com/mifos/feature/client/clientDetails/presentation/ClientDetailsScreen.kt

```diff
@@ -0,0 +1,978 @@
+
+
+    var clientNotFoundError by rememberSaveable { mutableStateOf(false) }
+    var showMenu by remember { mutableStateOf(false) }
+    var showSelectImageDialog by remember { mutableStateOf(false) }
+    var imageUri by remember { mutableStateOf<Uri?>(null) }
+    val snackbarHostState = remember { SnackbarHostState() }
+
+
+    val file = context.createImageFile()
+    val cameraImageUri = FileProvider.getUriForFile(
+        Objects.requireNonNull(context),
+        "com.mifos.mifosxdroid" + ".provider",
+        file
+    )
+    val permissionState = rememberMultiplePermissionsState(
+        permissions = listOf(
+            Manifest.permission.CAMERA
+        )
+    )
+
+    val galleryLauncher = rememberLauncherForActivityResult(
+        contract = ActivityResultContracts.GetContent(),
+        onResult = { uri ->
+            uri?.let {
+                imageUri = it
+                val bitmap = context.contentResolver.openInputStream(uri).use { stream ->
+                    BitmapFactory.decodeStream(stream).asImageBitmap().asAndroidBitmap()
+                }
+                showSelectImageDialog = false
+                clientDetailsViewModel.saveClientImage(clientId, bitmap)
+            }
+        }
+    )
+
+
+    val cameraLauncher =
+        rememberLauncherForActivityResult(contract = ActivityResultContracts.TakePicture()) { status ->
+            if (status) {
+                imageUri = cameraImageUri
+                val bitmap = context.contentResolver.openInputStream(cameraImageUri).use { stream ->
+                    BitmapFactory.decodeStream(stream).asImageBitmap().asAndroidBitmap()
+                }
+                showSelectImageDialog = false
+                clientDetailsViewModel.saveClientImage(clientId, bitmap)
+            }
+        }
+
+    LaunchedEffect(key1 = true) {
+        clientDetailsViewModel.loadClientDetailsAndClientAccounts(clientId)
+    }
+
+    when (state) {
+
+        is ClientDetailsUiState.ShowClientImageDeletedSuccessfully -> {
+            LaunchedEffect(key1 = true) {
+                snackbarHostState.showSnackbar(message = "Client Image Deleted Successfully")
+            }
+        }
+
+        is ClientDetailsUiState.ShowUploadImageSuccessfully -> {
+            LaunchedEffect(key1 = state.response) {
+                snackbarHostState.showSnackbar(message = "Client Image Uploaded successfully")
+            }
+        }
+
+        is ClientDetailsUiState.ShowError -> {
+            when (state.message == "null") {
+                true -> {
+                    clientNotFoundError = true
+                }
+
+                false -> {
+                    LaunchedEffect(key1 = state.message) {
+                        snackbarHostState.showSnackbar(message = state.message)
+                    }
+                }
+            }
+        }
+
+        ClientDetailsUiState.Empty -> {
```

Comment:
why do we need an Empty state?

Reply:
We are using MutableStateFlow which required initial value thats why i have used empty data object